index.d.ts

```diff
@@ -7,12 +7,26 @@ type TestComponent = {
   mount(): Promise<void>;
 };
 
-export = kompot;
+type Provider<P> = {
+  component: React.ComponentType<P>;
+  props: P;
+  children: React.ComponentType<any>[];
+}
+
+export = kompot; 
 export as namespace kompot;
 
 declare namespace kompot {
-  function spy(id: string, getReturnValue?: (args: any[]) => any, stringifyArgs?: (args: any[]) => string[]): any
+  function spy(id: string, getReturnValue?: (args: any[]) => any, stringifyArgs?: (args: any[]) => string[]): () => any;
+  function spyOn(obj: object, methodName: string, spyId: string, stringifyArgs?: (args: any[]) => string[]): void;
   function kompotRequire(pathToComponent: string): { [key: string]: TestComponent };
   function init(): void;
-  function expect<T>(value: T): jest.Matchers<T> & Detox.Expect<T> & {notToHaveBeenCalled(): Promise<void>}
+  function expect<T>(value: T): jest.Matchers<T> & Detox.Expect<T> & {notToHaveBeenCalled(): Promise<void>};
```

Comment:
kompot expect supports also expect spies

Reply:
exept for `notToHAveBeenCalled` all of the functions are aligned with the `jest.Matchers` type (TypeScript doesn't care about the runtime evaluation) so this should suffice for the typings.

Either that or I simply don't understand what you mean by `expect spies`

Reply:
something like this:
```javascript
      await expect(global.spy('fetch')).toHaveBeenCalledWith('wow')
```

Reply:
but you are right, I forgot that I used jest api...so it will probably work

Reply:
These types should cover this case